Sources/Stats/mc_stats.cc

```diff
@@ -141,21 +143,22 @@ Stats Statistics(Eigen::Ref<const Eigen::VectorXcd> values,
                    // chain.
       return std::make_pair(NaN, NaN);
     }
-    // TODO: Are the coefficients here correct??
-    return std::make_pair(global_var[0] / static_cast<double>(m - 1),
+
+    return std::make_pair(global_var[0] / static_cast<double>(m),
                           global_var[1] / static_cast<double>(m));
   }();
 
   if (!std::isnan(var.first) && !std::isnan(var.second)) {
     const auto t = var.first / var.second;
-    auto correlation = 0.5 * (static_cast<double>(n) * t * t - 1.0);
-    if (correlation < 0.0) correlation = NaN;
+    auto correlation = 0.5 * (t * static_cast<double>(n) - 1.0);
+    if (correlation < 0.0) correlation = 0;
     const auto R =
         std::sqrt(static_cast<double>(n - 1) / static_cast<double>(n) +
                   var.first / var.second);
-    return {mean, std::sqrt(var.first), var.second, correlation, R};
+    return {mean, std::sqrt(var.first / static_cast<double>(m)), var.second,
```

Comment:
@gcarleo you're now dividing by `m^2` (once during computation of `var` and the second time here). It seems wrong to me. Are you sure this is correct?

Reply:
yes, the second factor of m comes because we want the error on the mean, this is consistent... otherwise if you don't divide you would get the variance, not the error on the mean

Reply:
the thing that is still slightly wrong is that global_var[2] should contain the variance over all samples (i.e. merging all chains on all cores) whereas now it contains the average variance